### ISSUE TITLE
Use new DiagnosticId attributes in obsolete APIs

### ISSUE THREAD

**Issue by captainsafia** [dotnet/aspnetcore#29513]
From https://github.com/dotnet/aspnetcore/issues/27529#issuecomment-729878872.

> We may also want to use the new `[Obsolete]` `DiagnosticId` property. The dotnet/runtime team is using this extensively. See https://github.com/dotnet/runtime/blob/329f0dbfa3a164ee2e3bf9f75b7641165995c799/src/libraries/Common/src/System/Obsoletions.cs and e.g. https://github.com/dotnet/runtime/blob/329f0dbfa3a164ee2e3bf9f75b7641165995c799/src/libraries/System.Security.Permissions/src/System/Security/SecurityManager.cs#L10

This would allow us to create a unified experience for documenting obsolete APIs in the repo.

**Comment by halter73** [comment]
Here's the document describing the runtime process: https://github.com/dotnet/runtime/blob/472050d35cbbfb90c2d0372c97bbc3d271c5e68a/docs/project/list-of-diagnostics.md

**Comment by mkArtakMSFT** [comment]
@captainsafia I don't think this is infra work. I'm going to move it to the area-networking, as that's where most of the APIs are.
